### DISCUSSION TITLE
Bolt12 single offer with a wildcard DNS entry

### DISCUSSION THREAD

**Discussion by sixside** [ACINQ/phoenixd#100] [Ideas]
Future feature request: A way to associate a BOLT12 offer with a wildcard DNS entry so that a host could process inbound lightning payments to *@example.com and upon receipt of payment process it accordingly by crediting the dynamic user account with the funds received.

Example: someone sends 1000 sats to 123456username@example.com Phoenixd receives the payment, and triggers a webhook that then credits 123456username with 999 sats (1000 minus a 1sat fee according to internal biz logic) in the internal database of example.com's system.

See: https://x.com/TheBlueMatt/status/1813604927954186700

**Comment by t-bast** [comment]
This would need more thinking about the architecture to correctly expose this: ideally `phoenixd` would *not* internally manage such a multi-account system, but it could tag invoices with the corresponding user name (if correctly provided by the sender in the `invoice_request` or in the payment onion) and include that information in the payment event. This way an external app/plugin could listen to payment events and credit the right user account.
